### PR TITLE
Authentication configuration form

### PR DESCRIPTION
**Pull request by jkaster**
The stand-alone (non-extension) API Explorer uses OAuth and CORS requests for *TryIt* functionality. The configuration values for the API and Auth endpoints are either stored in `localStorage` or `sessionStorage`. This spike implements the basic functionality.

For the stand-alone API Explorer TryIt functionality:
- `ConfigForm` appears instead of the `RequestForm` (aka `TryIt`) until auth is configured
- `ConfigDialog` appears if the gear icon to the right of the TryIt button is clicked. This dialog can update or remove auth configuration.
- Both incarnations of the form are very ugly, but they work and styling can be another PR

## ConfigForm
image

## ConfigDialog
image

## Form validation
image

### CODE REVIEW

File: packages/try-it/src/components/ConfigDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  DialogManager,
+  DialogContent,
+  Icon,
+  IconNames,
+} from '@looker/components'
+import { ConfigForm } from './ConfigForm'
+
+interface ConfigDialogProps {
+  icon?: IconNames
+  setHasConfig?: Dispatch<boolean>
```
Comment (josephaxisa) [inline packages/try-it/src/components/ConfigDialog.tsx:38]:
```suggestion
  /** Icon to use for config dialog */
  icon?: IconNames
  /** A set state callback fn used to set a hasConfig flag based indicating whether OAuth config details are present */
  setHasConfig?: Dispatch<boolean>
```

File: packages/try-it/src/components/ShowResponse/ShowResponse.tsx [hunk 2]
```diff
@@ -63,7 +63,7 @@ export const ShowResponse: FC<ShowResponseProps> = ({
       <Heading as="h4">{`${verb || ''} ${path || ''} ${response.statusCode}: ${
         response.contentType
       }`}</Heading>
-      {pickedHandler.component(response)}
+      {pickedHandler && pickedHandler.component(response)}
```
[the remark this answers is not in the document]
Comment (jkaster) [reply packages/try-it/src/components/ShowResponse/ShowResponse.tsx:66]: I just got tired of the lint warning

File: packages/try-it/src/components/RequestForm/formUtils.tsx [hunk 3]
```diff
@@ -27,11 +27,11 @@ import React, { BaseSyntheticEvent, Fragment } from 'react'
 import {
   FieldText,
   FieldToggleSwitch,
-  InputDate,
   Label,
   FieldCheckbox,
 } from '@looker/components'
 
+import { InputDate } from '@looker/components/lib/InputDate'
```
[the remark this answers is not in the document]
Comment (jkaster) [reply packages/try-it/src/components/RequestForm/formUtils.tsx:34]: It's evidently not a top-level export. Sounds like a bug we need to report.

File: packages/try-it/src/components/configUtils.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  try {
+    const result = new URL(url)
+    return result.origin
+  } catch {
+    return ''
+  }
+}
+
+export const validLocation = (location: string) => {
+  return ['local', 'session'].includes(location.toLocaleLowerCase())
```
[the remark this answers is not in the document]
Comment (jkaster) [reply packages/try-it/src/components/configUtils.ts:103]: Sure. It's not the code I want, but evidently the code I want doesn't exist.

File: packages/try-it/src/components/ConfigForm.tsx [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+          description="Configuration storage location"
+          label="Save to"
+          name="location"
+          options={storageOptions}
+          defaultValue={fields.location}
+          onChange={handleLocationChange}
+          inline
+          required
+        />
+        {!dialogue && (
```
[the remark this answers is not in the document]
Comment (jkaster) [reply packages/try-it/src/components/ConfigForm.tsx:192]: That's from when a Cancel button was being discussed. If we're not going to surface a Cancel button (would only be for the dialog) I can remove that.

File: packages/try-it/src/components/RequestForm/formUtils.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (josephaxisa) [reply packages/try-it/src/components/RequestForm/formUtils.tsx:34]: they changed that here: https://github.com/looker-open-source/components/blob/master/CHANGELOG.md

[the remark this answers is not in the document]
Comment (jkaster) [reply packages/try-it/src/components/RequestForm/formUtils.tsx:34]: Cool. So not a bug but I'd like to understand why.

**Pull request merged**